apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplRetryTest.cs

```diff
@@ -0,0 +1,459 @@
+            service => service.Projects.HmacKeys.Update(new HmacKeyMetadata { ProjectId = "project", AccessId = "access", ETag = "check" }, "project", "access"), 
+            client => client.UpdateHmacKey(new HmacKeyMetadata { ProjectId = "project", AccessId = "access", ETag = "check" }));
+
+        [Fact]
+        public void UpdateHmacKey_RetryThenFail() => RetryThenFailHelper(
+            service => service.Projects.HmacKeys.Update(new HmacKeyMetadata { ProjectId = "project", AccessId = "access", ETag="check" }, "project", "access"), 
+            client => client.UpdateHmacKey(new HmacKeyMetadata { ProjectId = "project", AccessId = "access" , ETag = "check"}));
+
+        #endregion
+
+        #region Notifications test cases 
+
+        [Fact]
+        public void ListNotifications_NoRetry() =>
+            NoRetryHelper(service => service.Notifications.List("bucket"), client => client.ListNotifications("bucket").Count());
+
+        [Fact]
+        public void ListNotifications_RetryOnce() =>
+            RetryOnceHelper(service => service.Notifications.List("bucket"), client => client.ListNotifications("bucket").Count());
+
+        [Fact]
+        public void ListNotifications_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Notifications.List("bucket"), client => client.ListNotifications("bucket").Count());
+
+        [Fact]
+        public void GetNotification_NoRetry() =>
+            NoRetryHelper(service => service.Notifications.Get("bucket", "notification"), client => client.GetNotification("bucket", "notification"));
+
+        [Fact]
+        public void GetNotification_RetryOnce() =>
+            RetryOnceHelper(service => service.Notifications.Get("bucket", "notification"), client => client.GetNotification("bucket", "notification"));
+
+        [Fact]
+        public void GetNotification_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Notifications.Get("bucket", "notification"), client => client.GetNotification("bucket", "notification"));
+
+        [Fact]
+        public void DeleteNotification_NoRetry() =>
+            NoRetryHelper(service => service.Notifications.Delete("bucket", "notification"), client => client.DeleteNotification("bucket", "notification"));
+
+        [Fact]
+        public void DeleteNotification_RetryOnce() =>
+            RetryOnce(service => service.Notifications.Delete("bucket", "notification"), client => client.DeleteNotification("bucket", "notification"), "check");
+
+        [Fact]
+        public void DeleteNotification_RetryThenFail() =>
+            RetryThenFail(service => service.Notifications.Delete("bucket", "notification"), client => client.DeleteNotification("bucket", "notification"), "check");
+
+        #endregion
+
+        #region Helper Methods
+
+        /// <summary>
+        /// Helper method to check in case of non retriable error codes
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
```

Comment:
Please either add useful comments here or remove the elements - likewise for the other XML comments below.